### PR TITLE
Add Feedback Form

### PR DESCRIPTION
**Pull request by jgarber623-gov**
## Checklist

I have…

- [x] run the application locally (`bin/rails server`) and verified that my changes behave as expected.
- [x] run static code analysis (`bin/rubocop`) and vulnerability scan (`bin/brakeman`) against my changes.
- [x] run the test suite (`bin/rake spec`) and verified that all tests pass.
- [x] summarized below my changes and noted which issues (if any) this pull request fixes or addresses.
- [x] thoroughly outlined below the steps necessary to test my changes.
- [x] attached screenshots illustrating relevant behavior before and after my changes.

## Summary of Changes

This pull request adds a user-facing Feedback Form (available at `/feedback`). The implementation uses Rails' [Action Mailer](http://guides.rubyonrails.org/action_mailer_basics.html) and a single view to generate a plaintext email including the user's feedback.

## Testing

To verify the changes proposed in this pull request…

1. clone this repo,
1. `git checkout add-feedback-form`,
1. set up development dependencies according to `CONTRIBUTING.md`,
1. run `brew services run mailhog` (if it's not already running),
1. run `bin/rails server`,
1. load up http://localhost:3000/feedback, fill out and submit the form, and
1. load up http://localhost:8025 and observe a received email!

## Screenshots

<details>
  <summary>Initial Screen</summary>

  narrow

  wide
</details>

<details>
  <summary>Thank You Screen</summary>

  thanks-narrow

  thanks-wide
</details>

### CODE REVIEW

File: app/helpers/feedback_mailer_helper.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+module FeedbackMailerHelper
+  # rubocop:disable Metrics/AbcSize
+  def typeset(str, max_length = 80)
```
Comment (aileendds) [inline app/helpers/feedback_mailer_helper.rb]: Why do this?

File: app/views/feedback/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    <div class="usa-width-one-half">
+      <%= form_with scope: :feedback, url: feedback_path, class: 'usa-form', id: 'feedback-form' do |form| %>
+        <%= form.label :topic, 'Which category best describes your feedback?' %>
+        <%= form.select :topic, options_for_select(@topics, selected: params[:topic]), { prompt: '- Select -' }, { id: :feedback_topic, required: false } %>
+
+        <%= form.label :message, 'Feedback' %>
+        <%= form.text_area :message, id: :feedback_message, required: false, value: params[:message] %>
+
+        <div aria-hidden="true" class="honeybucket-container" hidden>
+          <%= form.label :honeybucket, 'What kind of robot are you?' %>
```
Comment (aileendds) [inline app/views/feedback/index.html.erb]: 🤖

**Comment by jamesatheyDDS** [comment]
@jgarber623-gov Did you test on IE11 and Edge?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -54,17 +57,42 @@ This script will install the dependencies specified in the project's [Gemfile][g
...
 
 ```
 export GOOGLE_MAPS_API_KEY=<your api key here>
 ```
+
 Enable the Google Maps Geocoding API by going [here](https://console.developers.google.com/apis/api/geocoding_backend).
 
+#### Encrypted Seed Data
+
 To work with encrypted seed data, you need to know the secret `SEEDS_ENC_KEY` variable and set it in the `.env` file. This was originally generated once during the `bin/setup` process. Once set, run `bin/rails db:setup` to load the encrypted data.
```
Comment (jamesatheyDDS) [inline CONTRIBUTING.md]: you need to know the secret `SEEDS_ENC_KEY` _and_ `SEEDS_ENC_IV`.

File: app/views/feedback/thanks.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<%= render partial: 'shared/contextual_header', locals: { page_title: @page_title, parent_pages: [] } %>
+
+<div class="usa-section main-section">
+  <h1><%= @page_title %></h1>
+
+  <div class="feedback-thanks-container">
+    <%= image_tag 'icons/feedback.svg', alt: '', role: 'presentation' %>
+
+    <p><b>Thanks for your feedback!</b></p>
+    <p>Your feedback has been sent to USTRANSCOM so that we can continue to improve the military relocation experience.</p>
```
Comment (jamesatheyDDS) [inline app/views/feedback/thanks.html.erb]: Bike-shedding some here, but what do you think about rewording this as:

> Your feedback has been sent to USTRANSCOM. Thank you for your input as we continue to improve the military relocation experience.

Comment (jgarber623-gov) [reply app/views/feedback/thanks.html.erb]: Sounds good to me!

**Comment by jamesatheyDDS** [comment]
@jgarber623-gov I'll rephrase my question - would it be helpful if someone with a working Windows VM running IE 11 were to test this PR?

**Comment by jgarber623-gov** [comment]
> would it be helpful if someone with a working Windows VM running IE 11 were to test this PR?

@jamesatheyDDS That would be most helpful! 😄

**Comment by jamesatheyDDS** [comment]
IE11:

virtualbox_msedge - win10_14_03_2018_12_20_31

MSEdge:

virtualbox_msedge - win10_14_03_2018_12_19_13

**Pull request merged**